dataflow/src/ops/grouped/mod.rs

```diff
@@ -222,15 +223,23 @@ where
                         }
                     };
 
-                    // TODO: take advantage of rs possibly being a Cow::Owned to avoid cloning
-                    // further down.
-                    let old = rs.get(0);
+                    let old = match rs {
+                        RecordResult::Owned(rows) => {
+                            rows.into_iter().next().map(|rs| Cow::from(rs))
+                        }
+                        RecordResult::Borrowed(ref rows) => {
+                            rows.get(0).as_ref().map(|rs| Cow::from(&rs[..]))
+                        }
```

Comment:
Given that we'll use this pattern a lot, it'd be nice if there was a handy utility method for it. Maybe something on `RecordResult` that yields an `Iterator<Item=Cow<[DataType]>>`?

Reply:
I think this can now also use `.into_iter()`